src/snarks/groth16/mpc_utils.cpp

```diff
@@ -0,0 +1,215 @@
+#include "mpc_utils.hpp"
+
+#include "evaluation_from_lagrange.hpp"
+#include "multi_exp.hpp"
+
+#include <libff/algebra/scalar_multiplication/multiexp.hpp>
+
+using namespace libsnark;
+
+namespace libzeth
+{
+
+using ppT = libff::default_ec_pp;
+using Fr = libff::Fr<ppT>;
+using G1 = libff::G1<ppT>;
+using G2 = libff::G2<ppT>;
+
+// -----------------------------------------------------------------------------
+// mpc_layer1
+// -----------------------------------------------------------------------------
+
+mpc_layer1::mpc_layer1(
```

Comment:
With `mpc_layer1` are you referring to the `Round 1` of the mpc (namely "Powers of Tau", 6.1 in the paper) + The result of the computation of the linear combinations between the rounds (6.2 in the paper)?

Reply:
This holds only the linear combination data.  It does need a better name.

The paper indexes the layers $C_i$ and $L_i$ and uses phrases such as "We fix a layer $l$ and denote $C=C_l$ and $L=L_l$ ...".  So this is the output from the "second layer", or $L_1$, or the "Linear combinations between phases".

How about `mpc_layer_L1` or `mpc_linear_combination`?  Let me know what you think would be a good name.

Reply:
I see, thanks for clarifying.
I think `mpc_linear_combination` relates well to the "Linear combinations between phases", so this one looks good.

Reply:
I see thanks for clarifying!
Maybe `srs_mpc_layer_L1` (or output_mpc_layer_L1) then?

Reply:
OK.  I'll go for `srs_mpc_layer_L1` then.  Thanks.